_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,252 @@
+---
+id: j7zzqr
+name: ARIA role is permitted for the element.
+rule_type: atomic
+description: |
+  This rule checks that explicit WAI-ARIA roles are allowed for the element they are specified on.
+accessibility_requirements:
+  html-aria:docconformance:
+    title: ARIA in HTML, 4. Document conformance requirements for use of ARIA attributes in HTML
```

Comment:
Sorry, this is going to be very nit picky.

> The third cell in each row defines the ARIA role values and aria-* attributes which authors MAY specify on the element

I don't think the way you're interpreting this requirement is right. Or quite possibly, I think the requirement is written incorrectly. MAY in RFC2119 speak does not create a prohibitive statement. You can use these roles on those elements, and browsers have to support it when you do. But that's not the same as saying authors are not allowed to use other roles. It would have to be something like "if an author specifies a role it MUST be one from the third column."

I think this is a solid rule, and we should definitely have it, but I'm a little reluctant to say this is a conformance requirement for ARIA in HTML. Even if that was the intended meaning, that's not what it says. I don't really want to hold this rule up over a technicality like this, but I do think something needs to be done about this.

Reply:
@WilcoFiers the sentence you quote needs to be put into the context of section 1 Author requirements fo ruse of ARIA in HTML

>Authors MUST NOT use the ARIA role and aria-* attributes in a manner that conflicts with the semantics described in the [4. Document conformance requirements for use of ARIA attributes in HTML](https://w3c.github.io/html-aria/#docconformance) and [4.2 Requirements for use of ARIA attributes in place of equivalent HTML attributes](https://w3c.github.io/html-aria/#docconformance-attr) tables. ...

per section 1, it's an Authors MUST NOT - where the Author MAYs specificy what it allowed.

if you think this is unclear because one has to piece together these different parts of the specification, then that's fair.  The spec can be updated to restate the author requirements more overtly in tandem with the text you quoted.